### PR TITLE
Feature/oki i18n error messages

### PR DESCRIPTION
**Pull request by mhicauber**
Proposition to I18N error messages

### CODE REVIEW

File: src/main/java/org/rutebanken/tiamat/versioning/ValidityUpdater.java [hunk 1]
```diff
@@ -37,34 +49,46 @@
...
+            String entityString = existingVersion.getNetexId() + " " + existingVersion.getVersion();
+            validateNewVersionDateAfter(VALIDATION_TO_DATE_AFTER_NEXT_VERSION_FROM_DATE, entityString, existingVersion.getValidBetween().getToDate(), newVersion.getValidBetween().getFromDate());
+            validateNewVersionDateAfter(VALIDATION_FROM_DATE_AFTER_NEXT_VERSION_FROM_DATE, entityString, existingVersion.getValidBetween().getFromDate(), newVersion.getValidBetween().getFromDate());
         }
 
         return newVersion.getValidBetween().getFromDate();
     }
 
+    /**
+     * FIXME : any way to have incoming TZ instead of relying on default TZ ?
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/versioning/ValidityUpdater.java]: ExportTimeZone is only used for dates being exported or serialized in messages like this, but shall never be used for parsing incoming data (hence, the name). Dates in incoming graphql requests are parsed in DateScalar. This parser requires timezone in the incoming datetime (not using export time zone). Therefore, the timezone is known, and could maybe be used to generate the exception message? Anyhow, the exception message will at the time of writing contain the correct datetime, but maybe represented in another timezone.

File: src/main/java/org/rutebanken/tiamat/config/Messages.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package org.rutebanken.tiamat.config;
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/config/Messages.java]: I see that the class MessageConfig is configuration. But this class responsibility is to retrieve messages. Should it be moved to pacakage org.rutebanken.tiamat.messages or similar?

File: src/main/java/org/rutebanken/tiamat/config/Messages.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.MessageSourceAccessor;
+import org.springframework.stereotype.Component;
+
+/**
+ * Util class to retrieve localized messages.
+ */
+@Component
+public class Messages {
```
Comment (csolem) [inline src/main/java/org/rutebanken/tiamat/config/Messages.java]: I'm wondering if the name of this class should be a bit more specific. LocalizedMessageRetriever or similar?

**Comment by csolem** [comment]
Had any change to review my comments yet?

**Comment by mhicauber** [comment]
Hi

No - had no time to address it yet, and it moved slowly to the bottom of my
todo list... Re-prioritizing it right now.

**Pull request closed without merging**